### PR TITLE
Fix Collector configuration issues

### PR DESCRIPTION
**Pull request by theletterf**
Fixes https://github.com/open-telemetry/opentelemetry.io/issues/1697 and adds some rewrites for clarity.

---

**Preview**: https://deploy-preview-3443--opentelemetry.netlify.app/docs/collector/configuration/

### CODE REVIEW

**Review by svrnm — COMMENTED** [review]
looks good overall, minor discussion/suggestion point around the endpoint URLs

File: content/en/docs/collector/configuration.md
Comment (theletterf) [reply content/en/docs/collector/configuration.md:362]: Yeah, I was unsure about which pattern to follow. I'm partial to using localhost everywhere if that's the most common use case in agent deployments. WDYT?

Comment (svrnm) [reply content/en/docs/collector/configuration.md:362]: Yeah, it might be confusing, so I would be OK with service.example.com:port

@open-telemetry/collector-approvers WDYT?

**Review by cartermp — APPROVED** [review]
Looks good. @open-telemetry/collector-approvers please take a look!

File: content/en/docs/collector/configuration.md
Comment (chalin) [inline content/en/docs/collector/configuration.md:506]: Could we drop the colons here? I think that it is sufficient to name the YAML key. WDYT?

File: content/en/docs/collector/configuration.md
Comment (chalin) [inline content/en/docs/collector/configuration.md:185]: I'd like to suggest that we drop the colon from the section name. The colon is an artifact of the YAML syntax. The section name is just the YAML key identifier. WDYT?

```suggestion
Receivers are configured in the `receivers` section. Many receivers come with
```

If you agree, this change will need to be applied elsewhere in the page.

File: content/en/docs/collector/configuration.md
Comment (chalin) [inline content/en/docs/collector/configuration.md:263]:
```suggestion
You can configure processors using the `processors` section of the Collector
```

File: content/en/docs/collector/configuration.md
Comment (chalin) [inline content/en/docs/collector/configuration.md:343]:
```suggestion
The `exporters` section contains exporters configuration. Most exporters
```

File: content/en/docs/collector/configuration.md
Comment (chalin) [inline content/en/docs/collector/configuration.md:640]:
```suggestion
To add a server authenticator to a receiver in the Collector, follow these steps:
```

Strange that Prettier didn't complain about the lone colon on its own line.

**Review by chalin — COMMENTED** [review]
Very nice update, thanks! ✨ 

See inline for suggested copyedits.

(This page uses blockquote syntax for lightweight side notes, which I think is fine for now.)

File: content/en/docs/collector/configuration.md
Comment (theletterf) [reply content/en/docs/collector/configuration.md]: Done! Switched to uppercase items so as to turn the list into merely structural, rather than YAML related.

**Comment by chalin** [comment]
@theletterf - could you rebase please? I can't from the interface apparently because of conflicts. Thanks!

File: content/en/docs/collector/configuration.md
Comment (chalin) [inline content/en/docs/collector/configuration.md:424]:
```suggestion
You can configure one or more connectors using the `connectors` section of the
```

File: content/en/docs/collector/configuration.md
Comment (chalin) [inline content/en/docs/collector/configuration.md:433]:
```suggestion
`pipelines` section. Notice that the connector acts as an exporter for traces
```

File: content/en/docs/collector/configuration.md
Comment (chalin) [inline content/en/docs/collector/configuration.md:472]:
```suggestion
You can configure extensions through the `extensions` section of the Collector
```

File: content/en/docs/collector/configuration.md
Comment (chalin) [inline content/en/docs/collector/configuration.md:525]:
```suggestion
- `traces` collects and processes trace data.
- `metrics` collects and processes metric data.
- `logs` collects and processes log data.
```

File: content/en/docs/collector/configuration.md
Comment (chalin) [inline content/en/docs/collector/configuration.md:520]:
```suggestion
Pipelines are defined in the `pipelines` subsection and can be of the following
```

File: content/en/docs/collector/configuration.md
Comment (chalin) [inline content/en/docs/collector/configuration.md:554]:
```suggestion
You can configure telemetry for the Collector itself in the `telemetry`
subsection inside the `service` section. Collector telemetry can be useful when
```

File: content/en/docs/collector/configuration.md
Comment (chalin) [inline content/en/docs/collector/configuration.md:560]:
```suggestion
The `logs` subsection lets you configure logs generated by the Collector. By
default, the Collector writes its logs to stderr with a log level of `INFO`. You
can also add static key-value pairs to all logs using the `initial_fields`
section. See the full list of `logs` options
```

File: content/en/docs/collector/configuration.md
Comment (chalin) [inline content/en/docs/collector/configuration.md:563]:
```suggestion
The `metrics` subsection lets you configure metrics generated by the Collector.
```

File: content/en/docs/collector/configuration.md
Comment (chalin) [inline content/en/docs/collector/configuration.md:565]: Would it make sense for this to be a (clickable) link rather than use code format?

File: content/en/docs/collector/configuration.md
Comment (chalin) [inline content/en/docs/collector/configuration.md:566]:
```suggestion
`metrics` subsection
```

File: content/en/docs/collector/configuration.md
Comment (chalin) [inline content/en/docs/collector/configuration.md:726]: Either use CFSSL or put in code font, e.g.:

```suggestion
Install [`cfssl`](https://github.com/cloudflare/cfssl) and create the following
```

**Review by chalin — COMMENTED** [review]
A few more copyedits.

File: content/en/docs/collector/configuration.md
Comment (theletterf) [reply content/en/docs/collector/configuration.md:565]: I think the link checker doesn't like it. Any way of adding `localhost` as an exception?

File: content/en/docs/collector/configuration.md
Comment (chalin) [inline content/en/docs/collector/configuration.md:565]: With the following change the link checker will be ok:

```suggestion
for scraping at <http://localhost:8888/metrics>. See the full list of options for the
```

File: content/en/docs/collector/configuration.md
Comment (chalin) [reply content/en/docs/collector/configuration.md:565]: See my followup suggestion. Marking this as resolved.

File: content/en/docs/collector/configuration.md
Comment (theletterf) [reply content/en/docs/collector/configuration.md:565]: Thanks!

**Comment by chalin** [comment]
This closes #1697, right?

**Comment by theletterf** [comment]
@chalin That's right! This can be merged (the other for Java servers requires waiting for legal approval from my employer).

**Comment by cartermp** [comment]
@open-telemetry/collector-approvers ping! Please take a look

**Comment by theletterf** [comment]
Could we merge and then fix in a separate PR once @open-telemetry/collector-approvers chime in?

File: content/en/docs/collector/configuration.md
Comment (chalin) [inline content/en/docs/collector/configuration.md:637]:
```suggestion
interested in developing a custom authenticator, see 
[Building a custom authenticator](../custom-auth).
```

File: content/en/docs/collector/configuration.md
Comment (chalin) [inline content/en/docs/collector/configuration.md:510]:
```suggestion
The `extensions` subsection consists of a list of all extensions to enable. For
```

File: content/en/docs/collector/configuration.md
Comment (chalin) [inline content/en/docs/collector/configuration.md:525]:
```suggestion
- `traces` collect and processes trace data.
- `metrics` collect and processes metric data.
- `logs` collect and processes log data.
```

File: content/en/docs/collector/configuration.md
Comment (chalin) [inline content/en/docs/collector/configuration.md:567]: As I mentioned in the other PR making changes to this section, I don't think we should be linking out to the source file. Especially because that [**config file has been deprecated**](https://github.com/open-telemetry/opentelemetry-collector/blob/50f013d0d242bf7b600eddf5cfe484e087c1de0c/CHANGELOG.md?plain=1#L1237)!

We should just list the "full options" (there aren't that many) here.

---

If you choose to keep the link for now, I'd at least suggest the following rewording to avoid a "here" link:

```suggestion
for scraping at <http://localhost:8888/metrics>. For the full list of options
for the `metrics` subsection, see the 
[`ServiceTelemetryMetrics struct`](https://github.com/open-telemetry/opentelemetry-collector/blob/7666eb04c30e5cfd750db9969fe507562598f0ae/config/service.go#L99-L111) comments.
```

**Review by chalin — COMMENTED** [review]
Another quick pass.

File: content/en/docs/collector/configuration.md
Comment (chalin) [reply content/en/docs/collector/configuration.md]: @theletterf et al.: I've created a followup issue to track further improvements to this point:

- #3514

**Review by chalin — APPROVED** [review]

**Comment by theletterf** [comment]
@svrnm This one can be merged, I think.

**Comment by svrnm** [comment]
> @svrnm This one can be merged, I think.

We need approval from someone from @open-telemetry/collector-approvers

File: content/en/docs/collector/configuration.md
Comment (codeboten) [inline content/en/docs/collector/configuration.md:264]: The line about requiring security settings could use a link to either an example or be removed. In theory any components may require security settings in order to function :)

File: content/en/docs/collector/configuration.md
Comment (codeboten) [inline content/en/docs/collector/configuration.md:255]: Not seeing a note in here about the ordering of processor matters, it's possible i missed it though

**Review by codeboten — COMMENTED** [review]
Thanks for the improvements! Just a couple of comments

File: content/en/docs/collector/configuration.md
Comment (theletterf) [reply content/en/docs/collector/configuration.md:264]: Always in for _less_ docs when it makes sense. :) Removing.

File: content/en/docs/collector/configuration.md
Comment (theletterf) [reply content/en/docs/collector/configuration.md:255]: Added in line 258.

**Comment by theletterf** [comment]
@codeboten Changes applied!

**Comment by svrnm** [comment]
@theletterf can you take a look at the conflict? I think this is related to some changes to the images I introduced yesterday, so it should not have any impact on the content.

**Comment by theletterf** [comment]
@svrnm Should be solved!

**Review by codeboten — APPROVED** [review]
Thanks @theletterf!

**Pull request merged**
